src/azure-cli/azure/cli/command_modules/appconfig/_kv_helpers.py

```diff
@@ -416,19 +429,19 @@ def __read_kv_from_app_service(cmd, appservice_account, prefix_to_add="", conten
             cmd, resource_group_name=appservice_account["resource_group"], name=appservice_account["name"], slot=slot)
         for item in settings:
             key = prefix_to_add + item['name']
-            if validate_import_key(key):
+            value = item['value']
+            if validate_import_key(key) and not value.strip().lower().startswith(AppServiceConstants.APPSVC_CONFIG_REFERENCE_PREFIX.lower()):  # Exclude app configuration references.
```

Comment:
Can we add a warning log for every AppConfig reference that we are skipping during import? Users may not be expecting it so its better to be more explicit.

Reply:
Update has been pushed for this. Kindly let me know if the wording works.